repository/scheduler.go

```diff
@@ -0,0 +1,234 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/ttab/elephant-api/repository"
+	"github.com/ttab/elephantine"
+	"github.com/ttab/elephantine/pg"
+)
+
+// SchedulerMaxPollInterval is the maximum interval for looking up the next
+// scheduled documents for publishing. This should be tweaked upwards when we
+// have a good signal for when scheduling data has changed. That would be
+// whenever a document enters the "withheld" workflow state or the publish time
+// is set or changed on an assignment.
+//
+// This controls how quickly new scheduled articles are discovered, it does not
+// reflect the resolution at which publishing can be scheduled, or how often the
+// scheduler can act on set times.
+const SchedulerMaxPollInterval = 1 * time.Minute
+
+// SchedulerRetryWindown is the time after a planned publish that the scheduler
+// will attempt to act on it. After the retry window has passed the document
+// will still be in the withheld state, unless otherwise changed, but no
+// attempts will be made to publish it.
```

Comment:
To make a bit more clear I would adjust the first sentence like so:
```suggestion
// SchedulerRetryWindow defines the duration after a planned publish time 
// during which the scheduler will attempt to publish the document.
// After the retry window has passed the document will still be in the withheld state,
// unless otherwise changed, but no attempts will be made to publish it.
```